pkg/types/ccipocr3/generic_types.go

```diff
@@ -108,7 +116,16 @@ func (c CCIPMsg) String() string {
 }
 
 type CCIPMsgBaseDetails struct {
-	ID          Bytes32       `json:"id"`
+	// ID is a unique identifier for the message, it should be unique across all chains.
+	ID string `json:"id"`
+	// SourceChain is the chain that the message originated from.
 	SourceChain ChainSelector `json:"sourceChain,string"`
-	SeqNum      SeqNum        `json:"seqNum,string"`
+	// SeqNum is an auto-incrementing sequence number for the message.
+	// NOTE: Sequence numbers are unique per chain. Meaning that the same sequence number can exist on multiple chains.
+	SeqNum SeqNum `json:"seqNum,string"`
+
+	// MsgHash is the hash of all the message fields. The hash is computed on the destination chain.
+	// This means that on the source chain we cannot compute the msg hash and in the off-chain side we need MsgHasher
+	// implementation for the destination chain.
+	MsgHash Bytes32 `json:"msgHash"` // computed
```

Comment:
The `// computed` comment reads to be like we should not be expecting to receive this field over the wire from encoded JSON, but that is not actually the case, right? Can we update this to be clear about when this is computed? The long comment reads like it happens on-chain and cannot happen off-chain, but that doesn't make sense to me either. Is it a particular kind of hash?

Reply:
> comment reads to me like we should not be expecting to receive this field over the wire from encoded JSON, but that is not actually the case, right?

that's the case, i'll update the comment

Reply:
Oh, but then do we actually need the json tag? Or could it be:
```go
MsgHash Bytes32 `json:"-"`
```